onadata/libs/serializers/user_profile_serializer.py

```diff
@@ -167,7 +170,7 @@ def update(self, instance, validated_data):
         instance.user.save()
 
         # force django-digest to regenerate its stored partial digests
-        update_partial_digests(instance.user, params.get("password1"))
+        update_partial_digests(instance.user, password)
```

Comment:
There's an error here that needs a test, if you add a test that passes no `email` and no `password` in params this should execute with `password` set to `nil` and set the partial digests to a bad state.

I'd suggest writing that test and then getting it passing by adding if `password` here